Migrate `mock-observations` to `builtins` (#710)
Green-Software-Foundation/if: src/builtins/mock-observations/README.md

### Authentication

N/A

### Inputs

The plugin's `global-config` section in the manifest file determines its behaviour.
'inputs' section is ignored.

### Typescript Usage

```typescript
const mockObservations = MockObservations({
  'timestamp-from': '2023-07-06T00:00',
  'timestamp-to': '2023-07-06T00:10',
  duration: 60,
  components: {
    'instance-type': 'A1',
  },
  generators: {
    common: {
      region: 'uk-west',
    },
  },
});
const result = await mockObservations.execute([]);
```

### manifest Example

IF users will typically call the plugin as part of a pipeline defined in a `manifest` file. In this case, instantiating the plugin is handled by `ie` and does not have to be done explicitly by the user. The following is an example `manifest` that calls `mock-observation`:

```yaml
name: mock-observation-demo
description: example invoking mock-observation plugin
tags:
initialize:
  outputs:
    - yaml
  plugins:
    mock-observations:
      kind: plugin
      method: MockObservations

>       path: 'builtins'

Reviewer: should be `builtin`

Author: ```suggestion
      path: 'builtin'
```

      global-config:
        timestamp-from: 2023-07-06T00:00
        timestamp-to: 2023-07-06T00:10
        duration: 60
        components:
          - instance-type: A1
          - instance-type: B1
        generators:
          common:
            region: uk-west
            common-key: common-val
          randint:
            cpu/utilization:
              min: 1
              max: 99
            memory/utilization:
              min: 1
              max: 99
tree:
  children:
    child:
      pipeline:
        - mock-observations
      inputs:
```

> You can run this example `manifest` by saving it as `./examples/manifests/test/mock-observation.yml` and executing the following command from the project root:

Author: ```suggestion
You can run this example `manifest` by saving it as `manifests/plugins/mock-observation.yml` and executing the following command from the project root:
```

```sh
npm i -g @grnsft/if

> npm i -g @grnsft/if-plugins

Reviewer: please, remove this row and update manifest path

Author: ```suggestion
```
